bin/extract-ORFs.py

```diff
@@ -64,19 +86,59 @@ def translations(read):
         def translations(read):
             return read.translations()
 
+    if kozakInfo or kozakOnly:
+        if aa:
+            # Should the code break in this case?
+            print('Did you pass NA sequences? Kozak consensuses can only'
+                  'be found in a nucleotide sequence.', file=sys.stderr)
+        else:
+            from dark.dna import findKozakConsensus
+
+        def writeToKozakOut(kozakread, kozakInfoFile):
+            """
+            Writes out information about a Kozak sequence stored in kozakread
+            to the file name given in kozakInfoFile.
+            """
+            kozakout = open(kozakInfoFile, 'a')
+            kozakout.write('Read ID: ' + kozakread.id)
+            kozakout.write('\n')
+            kozakout.write('Kozak sequence: ' + kozakread.sequence)
+            kozakout.write('\n')
+            kozakout.write('Offset of the Kozak sequence A: ' +
+                           str(kozakread.start + 6))
+            kozakout.write('\n')
+            kozakout.write('Quality of the Kozak consensus: ' +
+                           str(kozakread.kozakQuality) + ' %')
+            kozakout.write('\n')
+            kozakout.write('\n')
+            kozakout.close()
+
     for read in reads:
         try:
             for translation in translations(read):
-                for orf in translation.ORFs():
-                    # Omit open ORFs if they're unwanted.
-                    if ((orf.openLeft or orf.openRight) and not allowOpenORFs):
-                        continue
-
+                for orf in translation.ORFs(allowOpenORFs):
                     # Check the length requirements, if any.
                     if ((minORFLength is None or len(orf) >= minORFLength) and
-                            (maxORFLength is None or
-                                len(orf) <= maxORFLength)):
-                        write(orf.toString('fasta'))
+                       (maxORFLength is None or len(orf) <= maxORFLength)):
+                        # If Kozak consensus information is wanted:
+                        if kozakInfo and not aa:
+                            kozakreads = list(findKozakConsensus(read))
+                            for kozakread in kozakreads:
+                                writeToKozakOut(kozakread, kozakInfoFile)
+                            sys.stdout.write(orf.toString('fasta'))
```

Comment:
You should probably never need to call `sys.stdout.write`. Just use `print` (though here you'll need `end=''` to prevent the newline being added - my mis-design).

Reply:
done